mono/mini/method-to-ir.c

```diff
@@ -10605,8 +10610,42 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 				break;
 			}
 
-			*sp++ = handle_box (cfg, val, klass, context_used);
+			if (mono_class_is_nullable (klass) || mini_is_gsharedvt_klass (klass) || context_used) {
```

Comment:
I'm curious, why can't we emit box ops for those 3 cases? It would be nice to document the constraints.